### PR TITLE
P4 2426 list moves

### PR DESCRIPTION
**Pull request by jcavell**
Preliminary implementation of list moves

Clicking on the move type link takes you to list moves for that move type.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/move/MovesPage.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     * Returns the offset to be taken according to the underlying page and page size.
+     *
+     * @return the offset to be taken
+     */
+    override fun getOffset(): Long = offset
+
+    /**
+     * Returns the sorting parameters.
+     *
+     * @return
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/move/MovesPage.kt:34]: Docs a little scrappy...meh

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/move/MovesPage.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+     */
+    override fun getSort(): Sort = sort
+
+    /**
+     * Returns the [Pageable] requesting the next [Page].
+     *
+     * @return
+     */
+    override fun next(): Pageable {
+        TODO("Not yet implemented")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/move/MovesPage.kt:44]: Do we have a ticket number to include in the TODO's.  Just useful so can tie back to JIRA.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/MovesForMonthService.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                    sumByDouble { it.percentage },
+                    sumBy { it.volume },
+                    sumBy { it.volumeUnpriced },
+                    sumBy { it.totalPriceInPence }
+            )
+        }
+    }
+}
+
+fun endOfMonth(startDate: LocalDate): LocalDate = startDate.plusMonths(1).minusDays(1)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/MovesForMonthService.kt:66]: Make private and embed in class MovesForMonthService?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/MovesForMonthServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        assertThat(firstPage.last().moveId).isEqualTo("first50")
+        val secondPage = service.paginatedMovesForMoveType(Supplier.SERCO, MoveType.STANDARD, date, firstPage.nextPageable())
+        assertThat(secondPage.first().moveId).isEqualTo("second50")
+
+        val lastPage = service.paginatedMovesForMoveType(Supplier.SERCO, MoveType.STANDARD, date, secondPage.nextPageable())
+        assertThat(lastPage.numberOfElements).isEqualTo(1) // just one element
+        assertThat(lastPage.first().moveId).isEqualTo("lastOne")
+
+
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/MovesForMonthServiceTest.kt:67]: What happened to the formatting?  Fall asleep on the keyboard?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/MovesForMonthServiceTest.kt [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import java.time.LocalDate
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ActiveProfiles("test")
+@ContextConfiguration(classes = [TestConfig::class])
+ class MovesForMonthServiceTest(){
+
+    @MockBean
+    lateinit var moveQueryRepository: MoveQueryRepository
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/MovesForMonthServiceTest.kt:28]: Could perhaps add some `verify's` on the repo.  For example confirm the end of month logic is correct?

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Review by danielepolencic — APPROVED** [review]

**Pull request merged**